juriscraper/opinions/united_states/state/ill.py

```diff
@@ -1,61 +1,80 @@
-# Contact: webmaster@illinoiscourts.gov, 217-558-4490, 312-793-3250
-# History:
-#   2013-08-16: Created by Krist Jin
-#   2014-12-02: Updated by Mike Lissner to remove the summaries code.
-#   2016-02-26: Updated by arderyp: simplified thanks to new id attribute identifying decisions table
-#   2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
+# -*- coding: utf-8 -*-
+"""
+Contact: webmaster@illinoiscourts.gov, 217-558-4490, 312-793-3250
+ History:
+   2013-08-16: Created by Krist Jin
+   2014-12-02: Updated by Mike Lissner to remove the summaries code.
+   2016-02-26: Updated by arderyp: simplified thanks to new id attribute identifying decisions table
+   2016-03-27: Updated by arderyp: fixed to handled non-standard formatting
+   2021-11-02: Status validation based on Amendment to Rule 23 rulings
+               https://chicagocouncil.org/illinois-supreme-court-amendment-to-rule-23/
+"""
 
-from juriscraper.OpinionSite import OpinionSite
-from juriscraper.lib.string_utils import convert_date_string
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+from juriscraper.AbstractSite import logger
+from juriscraper.lib.html_utils import (
+    get_row_column_links,
+    get_row_column_text,
+)
+import datetime
+import re
 
 
-class Site(OpinionSite):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super(Site, self).__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.url = "http://www.illinoiscourts.gov/Opinions/recent_supreme.asp"
-        # Sometimes court forgets pdf link in 5th cell.
-        # So, only process rows with a link in 5th cell.
-        limiter = 'contains(./td[5]//a[1]/@href, "pdf")'
-        self.cell_path = '//table[@id="decisions"]//tr[' + limiter + "]/td[%d]"
+        self.url = (
+            "https://www.illinoiscourts.gov/top-level-opinions?type=supreme"
+        )
 
-    def _get_download_urls(self):
-        path = "%s//a[1]/@href" % (self.cell_path % 5)
-        return [href for href in self.html.xpath(path)]
+    def _process_html(self):
+        date_amend_rule_23 = datetime.date(2021, 1, 1)
+        for row in self.html.xpath("//table[@id='ctl04_gvDecisions']//tr"):
+            cells = row.xpath(".//td")
+            if len(cells) == 7:
```

Comment:
This looks like it could probably be flipped for better readability:

```suggestion
            if len(cells) != 7:
                continue
```

Reply:
Change done